### PR TITLE
[Gemma][Llama] fix bf16 min_dtype

### PR DESCRIPTION
**Pull request by usstq**
# What does this PR do?

`min_dtype` of torch.float32 type in Llama/Gemma's _update_causal_mask() will converted into bf16's `-infinity` by CPU plugin (due to design of https://www.felixcloutier.com/x86/vcvtneps2bf16 instruction).

this PR fixes it by using `torch.finfo(torch.bfloat16).min` to replace `torch.finfo(torch.float32).min`, they are very close in magnitude but won't cause `-infinity` issue.

```python
>>> torch.finfo(torch.float).min
-3.4028234663852886e+38
>>> torch.finfo(torch.bfloat16).min
-3.3895313892515355e+38
```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by eaidova** [comment]
@usstq , thanks, I prepared similar changes with only difference that it uses fp16 as input dtype as minimal for all supported devices #654

File: optimum/exporters/openvino/model_patcher.py [hunk 1]
```diff
@@ -288,6 +289,73 @@ def __exit__(self, exc_type, exc_value, traceback):
...
+def _update_causal_mask(self, attention_mask, input_tensor, cache_position):
+    if self.config._attn_implementation == "flash_attention_2":
+        if attention_mask is not None and 0.0 in attention_mask:
+            return attention_mask
+        return None
+
+    dtype, device = input_tensor.dtype, input_tensor.device
+    min_dtype = torch.finfo(dtype).min
+
+    # workaround CVS-137270
```
Comment (eaidova) [inline optimum/exporters/openvino/model_patcher.py:303]: I'm not sure that it is good idea to mention openvino jira ticket in external repo

File: optimum/exporters/openvino/model_patcher.py [hunk 2]
```diff
@@ -301,6 +369,27 @@ def __enter__(self):
                     rotary_emb.base ** (torch.arange(0, rotary_emb.dim, 2, dtype=torch.int64).float() / rotary_emb.dim)
                 )
 
+        self._model.model.original_update_causal_mask = self._model.model._update_causal_mask
+        self._model.model._update_causal_mask = types.MethodType(_update_causal_mask, self._model.model)
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        super().__exit__(exc_type, exc_value, traceback)
+        self._model.model._update_causal_mask = self._model.model.original_update_causal_mask
+        del self._model.model.original_update_causal_mask
```
[the remark this answers is not in the document]
Comment (usstq) [reply optimum/exporters/openvino/model_patcher.py]: changed!

File: optimum/exporters/openvino/model_patcher.py (hunk 1 above)
Comment (usstq) [reply optimum/exporters/openvino/model_patcher.py]: removed!

File: optimum/exporters/openvino/model_patcher.py [hunk 3]
```diff
@@ -288,7 +289,82 @@ def __exit__(self, exc_type, exc_value, traceback):
...
+    if self.config._attn_implementation == "flash_attention_2":
+        if attention_mask is not None and 0.0 in attention_mask:
+            return attention_mask
+        return None
+
+    dtype, device = input_tensor.dtype, input_tensor.device
+    min_dtype = torch.finfo(dtype).min
+
+    if dtype == torch.float32 or dtype == torch.float:
+        min_dtype = torch.finfo(torch.bfloat16).min
```
Comment (eaidova) [inline optimum/exporters/openvino/model_patcher.py:304]: what do you think about using torch.finfo(torch.float16) instead? We tested yesterday and sow that it also works without accuracy issue. gemma has 2b variant that can be fit to arm cpu with fp16, I think it also can get the same overflow on this platform (just for avoiding fixing that twice)

**Review by eaidova — APPROVED** [review]

File: optimum/exporters/openvino/model_patcher.py (hunk 3 above)
Comment (usstq) [reply optimum/exporters/openvino/model_patcher.py:304]: Good point, fp16 should be better!
